src/Compilers/Shared/BuildServerConnection.cs

```diff
@@ -95,9 +95,10 @@ internal static string GetRuntimeDirectoryOpt()
             BuildPathsAlt buildPaths,
             string keepAlive,
             string libEnvVariable,
-            CancellationToken cancellationToken)
+            CancellationToken cancellationToken,
```

Comment:
My strategy was to minimize the impact of the changes; i.e. methods could be invoked as they were prior to the change with no additional changes required, and behavior would remain the same if new parameters were not provided.

To that end, I decided to pass sharedCompilationId to this method as an optional parameter; optional parameters must appear at the end of the list. By making cancellationToken optional, the order could be changed, but this would likely still require a more extensive refactor. If you prefer, I can create an overload for this method that satisfies the requirement so that neither cancellationToken nor sharedCompilationId are optional. Please advise.